python/packages/azure-ai/agent_framework_azure_ai/_client.py

```diff
@@ -188,8 +192,19 @@ async def _get_agent_reference_or_create(
         """
         agent_name = self.agent_name or "UnnamedAgent"
 
-        # If no agent_version is provided, create a new agent
+        # If no agent_version is provided, either use latest version or create a new agent:
         if self.agent_version is None:
+            # Try to use latest version if requested and agent exists
+            if self.use_latest_version:
+                try:
+                    existing_agent = await self.project_client.agents.retrieve(agent_name)
+                    self.agent_name = existing_agent.name
+                    self.agent_version = existing_agent.versions.latest.version
+                    return {"name": agent_name, "version": self.agent_version, "type": "agent_reference"}
```

Comment:
```suggestion
                    return {"name": self.agent_name, "version": self.agent_version, "type": "agent_reference"}
```

Reply:
Should this be `self.agent_name`? We're updating `self.agent_name = existing_agent.name` above and could potentially use the wrong name.